In [14]:
import networkx as nx
import networkx as nx

In [15]:
def ler_grafo_de_arquivo(arquivo):
    grafo = nx.Graph()
    
    with open(arquivo, 'r') as f:
        for linha in f:
            # Supõe que os nós estão separados por espaço na mesma linha
            no1, no2 = linha.strip().split()
            grafo.add_edge(int(no1), int(no2))
    return grafo

In [16]:
def descrever_grafo(grafo):
    print("O grafo contém {} vértices e {} arestas".format(grafo.number_of_nodes(),
                                                     grafo.number_of_edges()))
    print("O grau de assortatividade do grafo é de {}".format(nx.degree_assortativity_coefficient(grafo)))
    print("O grafo possui {} elementos conectados".format(nx.number_connected_components(grafo)))
    coeficiente_gigante = max(nx.connected_components(grafo), key = len)
    print("O tamanho do coeficiente gigante é {}".format(len(coeficiente_gigante)))
    gcc = nx.average_clustering(grafo)
    # Imprima o GCC
    print("Coeficiente de Clusterização Global (GCC):", gcc)
    

# Primeira análise
## Rede: email-Eu-core

In [17]:
email_grafo = ler_grafo_de_arquivo('email-Eu-core.txt')
descrever_grafo(email_grafo)

O grafo contém 1005 vértices e 16706 arestas
O grau de assortatividade do grafo é de -0.010990490627931076
O grafo possui 20 elementos conectados
O tamanho do coeficiente gigante é 986
Coeficiente de Clusterização Global (GCC): 0.3993549664221539


# Segunda análise
## Rede: p2pGnutella24

In [18]:
p2p_grafo = ler_grafo_de_arquivo('p2p-Gnutella24.txt')
descrever_grafo(p2p_grafo)

O grafo contém 26518 vértices e 65369 arestas
O grau de assortatividade do grafo é de -0.00772826636376463
O grafo possui 11 elementos conectados
O tamanho do coeficiente gigante é 26498
Coeficiente de Clusterização Global (GCC): 0.005502501232433069


# Terceira análise
## Rede: facebook_combined

In [19]:
facebook_grafo = ler_grafo_de_arquivo('facebook_combined.txt')
descrever_grafo(facebook_grafo)

O grafo contém 4039 vértices e 88234 arestas
O grau de assortatividade do grafo é de 0.06357722918564943
O grafo possui 1 elementos conectados
O tamanho do coeficiente gigante é 4039
Coeficiente de Clusterização Global (GCC): 0.6055467186200876


# Quarta análise
## Rede: Wiki-Vote

In [20]:
wiki_grafo = ler_grafo_de_arquivo('Wiki-Vote.txt')
descrever_grafo(wiki_grafo)

O grafo contém 7115 vértices e 100762 arestas
O grau de assortatividade do grafo é de -0.0830524827001603
O grafo possui 24 elementos conectados
O tamanho do coeficiente gigante é 7066
Coeficiente de Clusterização Global (GCC): 0.14089784589308738


# Quinta análise
## Rede: Wiki-Rfa

In [21]:
cit_grafo = ler_grafo_de_arquivo('Cit-HepTh.txt')
descrever_grafo(cit_grafo)

O grafo contém 27770 vértices e 352324 arestas
O grau de assortatividade do grafo é de -0.03030462195847781
O grafo possui 143 elementos conectados
O tamanho do coeficiente gigante é 27400
Coeficiente de Clusterização Global (GCC): 0.31201949580622557


|   | Qtd vértices | Qtd arestas | degree assortativity coefficient | Qtd componente conectados | Tamanho do componente gigante | Coeficiente de Clustering |
|---|--------------|-------------|----------------------------------|---------------------------|-----|---------------------------|
| Rede 'email' | 1005 | 16706 | -0.0110 | 20 | 986 | 0.3993 |
| Rede 'p2p' | 26518 | 65369 | -0.0077 | 11 | 26498 | 0.0055 |
| Rede 'facebook' | 4039 | 88234 | 0.0636 | 1 | 4039 | 0.6055 |
| Rede 'wiki' | 7115 | 100762 | -0.0831 | 24 | 7066 | 0.1409 |
| Rede 'cit' | 27770 | 352324 | -0.0303 | 143 | 27400 | 0.3120 |